pkg/graveler/committed/merge.go

```diff
@@ -250,7 +250,7 @@ func (m *merger) handleBothRanges(sourceRange *Range, destRange *Range) error {
 			m.haveDest = m.dest.Next()
 		}
 
-	case bytes.Compare(sourceRange.MaxKey, destRange.MinKey) < 0: // source before dest
+	case range1BeforeRange2(sourceRange, destRange): // source before dest
```

Comment:
Suggestion: instead, add the following method to `Range`
```
func (r Range) Before(other Range) {
...
}
```

Also for the other methods you've added in this file.